Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs

```diff
@@ -215,6 +217,18 @@ public float TeleportDuration
         }
 
         [SerializeField]
+        [Tooltip("Profile for wiring up diagnostic assets.")]
+        private MixedRealityDiagnosticsProfile diagnosticsProfile;
+
+        /// <summary>
+        /// Active profile for diagnostic configuration
+        /// </summary>
+        public MixedRealityDiagnosticsProfile DiagnosticsProfile
```

Comment:
Do we want to add Diagnostics to the core managers?

Or should we make it an optional component?

Reply:
given the value to devs, core makes sense.

considering that it will likely be disabled for retail builds, i can see it being optional

Reply:
How are we disabling it for production builds?

Reply:
If it's left in production, the overhead is memory and binary size right? Without enabling the profile, there should be no work being done. 

Could add an `#if DEBUG` check, or the equivalent for whatever works in unity.

Reply:
`# UNITY_EDITOR`

Reply:
That's partly why I wondered if we should just register this as a `IMixedRealityComponent`

Reply:
I don't really see a _bad_ reason to allow people to enable this in production though. For some games there's fps, memory, and cpu indicators accessible to users to help them monitor what's going wrong if something does go wrong. While I don't think it's our job to tell devs what information is best for this, having this available as an easy entry point doesn't hurt. 

It's good to know we can disable it completely for release builds. I'm not sure on policy of should we, but my gut says to leave it open and reduce surface area later if needed.

Reply:
I'm going to resolve this and keep as is for now. Feel free to reopen if you feel there's more to talk about.